scripts-dev/test_postgresql.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# This script builds the Docker image to run the PostgreSQL tests, and then runs
+# the tests. It uses a dedicated tox environment so that we don't have to
+# rebuild it each time.
+
+# Command line arguments to this script are forwarded to "tox" and then to "trial".
+
+set -e
+
+# Build, and tag
+docker build docker/ -f docker/Dockerfile-pgtests -t synapsepgtests
+
+# Create a place to persist tox's environment so reruns are faster.
+# NB: I'm using a separate docker volume here, because you may have already run
+# "tox -e py36-postgres" on your system. You might not be using the distro used
+# in Dockerfile-pgtests, and so might be using a different CPython build! I don't
+# want there to be any accidental clashes
+docker volume create synapse-pg-test-tox
```

Comment:
Is there a benefit to explicitly creating the volume? I can't remember ever needing to do this and thought `docker run` creates volumes automatically.

However I also think that it's convenient to having a nearby path mounted as the volume, rather than having to go hunting in `/var/lib/docker` wherever it is, when you need to read the logs (the logs *are* deposited into `tox`'s environment, right? correct me if wrong).

Reply:
Yeah, it goes into tox's environment AFAICS.

Maybe we should create a new tox env called say `localtest-postgres`? (I don't want to re-use an existing one, since the container's Python interperter might differ from that on your local system)

There's also the fun and games of making sure that the files persisted to the host aren't written as root (grr).

Reply:
My thought was something like mounting `$(PWD)/.tox-postgres` or something to `/tox`? Anything sensible is fine, however, as long as people know where to look for the logs (so we should write down where that is in the contributing guide)

Reply:
> aren't written as root (grr).

hmm, sad. I forgot about that. I don't know how to deal with that.